src/files-and-uploads/data/utils.js

```diff
@@ -1,11 +1,54 @@
 import { InsertDriveFile, Terminal, AudioFile } from '@edx/paragon/icons';
 import { ensureConfig, getConfig } from '@edx/frontend-platform';
+import JSZip from 'jszip';
+import saveAs from 'file-saver';
 import FILES_AND_UPLOAD_TYPE_FILTERS from './constant';
 
 ensureConfig([
   'STUDIO_BASE_URL',
 ], 'Course Apps API service');
 
+export const getDownloadZipFolder = async (selectedRows, courseId, onError) => {
+  if (selectedRows.length > 1) {
+    const zip = new JSZip();
+    const date = new Date().toString();
+    const folder = zip.folder(`${courseId}-assets-${date}`);
+    const assetNames = [];
+    const assetFetcher = await Promise.all(
+      selectedRows.map(async (row) => {
+        const asset = row.original;
+        assetNames.push(asset.displayName);
+        try {
+          const res = await fetch(`${getConfig().STUDIO_BASE_URL}/${asset.id}`);
```

Comment:
How come this isn't called via a dispatch and lives in api.js?

Reply:
I didn't really think about moving it into thunks or api because the first iteration of download didn't require api fetches. Definitely makes more sense to move it and call it via dispatch.